samza-core/src/test/scala/org/apache/samza/job/local/TestProcessJob.scala

```diff
@@ -20,49 +20,128 @@
 package org.apache.samza.job.local
 
 import org.apache.samza.coordinator.JobModelManager
+import org.apache.samza.job.ApplicationStatus.{SuccessfulFinish, UnsuccessfulFinish}
+import org.apache.samza.job.CommandBuilder
 import org.junit.Assert._
 import org.junit.Test
-import org.apache.samza.job.ApplicationStatus
-import org.apache.samza.job.CommandBuilder
+
 import scala.collection.JavaConverters._
 
-class TestProcessJob {
-  @Test
-  def testProcessJobShouldFinishOnItsOwn {
+object TestProcessJob {
+
+  val OneSecondCommand = "sleep 1"
+  val TenSecondCommand = "sleep 10"
+  val SimpleCommand = "true"
+  val FailingCommand = "false"
+  val BadCommand = "bad-non-existing-command"
+
+  private def createProcessJob(command: String): ProcessJob = {
     val commandBuilder = new CommandBuilder {
-      override def buildCommand = "sleep 1"
+      override def buildCommand = command
+
       override def buildEnvironment = Map[String, String]().asJava
     }
-    val coordinator = new MockJobModelManager()
-    val job = new ProcessJob(commandBuilder, coordinator)
-    job.submit
-    job.waitForFinish(999999)
+    new ProcessJob(commandBuilder, new MockJobModelManager)
   }
 
-  // TODO: fix in SAMZA-1261
-  // @Test
-  def testProcessJobKillShouldWork {
-    val commandBuilder = new CommandBuilder {
-      override def buildCommand = "sleep 999999999"
-      override def buildEnvironment = Map[String, String]().asJava
-    }
-    val coordinator = new MockJobModelManager()
-    val job = new ProcessJob(commandBuilder, coordinator)
-    job.submit
-    job.waitForFinish(500)
-    job.kill
-    job.waitForFinish(999999)
-    assertTrue(coordinator.stopped)
-    assertEquals(ApplicationStatus.UnsuccessfulFinish, job.waitForFinish(999999999))
+  private def getMockJobModelManager(processJob: ProcessJob): MockJobModelManager = {
+    processJob.jobModelManager.asInstanceOf[MockJobModelManager]
+  }
+}
+
+class TestProcessJob {
+
+  import TestProcessJob._
+
+  @Test
+  def testProcessJobShouldFinishOnItsOwn: Unit = {
+    val processJob = createProcessJob(SimpleCommand)
+
+    val status = processJob.submit.waitForFinish(0)
```

Comment:
I agree we don't practically need to wait indefinitely, but I'd like to communicate to anyone reading the code that we're waiting to completion without any expectations about how long it should take. Specifying any other number will most probably have to be accompanied by a comment explaining that the specified value has no special resemblance. 

Waiting indefinitely here also helps secure the test against flakiness in case `SimpleCommand` is changed or if spawning the new process takes longer than expected for whatever reason, e.g. limited machine resources.

Reply:
I notices you use it in the tests. If something goes wrong, can the test hang?

Reply:
Yes, and I agree a hanging test is a concern. 

However, I think it should be the responsibility of the test runner to abort a unit test if it takes too long to execute because practically any piece of code is susceptible to hanging, e.g. due to an infinite loop, if something goes wrong (I'm assuming you mean if the code has a bug).

If you think this test in particular should be treated differently (maybe because our CI/CD infrastructure does not auto-kill long running tests?), I could add a timeout annotation to it so that when it fails, the error will clearly state the test has timed out, which would communicate our intent way better IMHO than an arbitrary `join`/`wait` timeout that elapses then causes a subsequent assertion to fail.